server/payload_manager.go

```diff
@@ -3,61 +3,442 @@ package server
 import (
 	"crypto/ecdsa"
 	"crypto/rand"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
 
+	"github.com/golang/protobuf/proto"
+
+	"github.com/status-im/status-go/account/generator"
+	"github.com/status-im/status-go/eth-node/keystore"
+	"github.com/status-im/status-go/images"
+	"github.com/status-im/status-go/multiaccounts"
 	"github.com/status-im/status-go/protocol/common"
+	"github.com/status-im/status-go/protocol/protobuf"
 )
 
-type Payload struct {
+// PayloadManager is the interface for PayloadManagers and wraps the basic functions for fulfilling payload management
+type PayloadManager interface {
+	Mount() error
+	Receive(data []byte) error
+	ToSend() []byte
+	Received() []byte
+	ResetPayload()
+}
+
+// PairingPayloadManagerConfig represents the initialisation parameters required for a PairingPayloadManager
+type PairingPayloadManagerConfig struct {
+	DB                             *multiaccounts.Database
+	KeystorePath, KeyUID, Password string
+}
```

Comment:
This might be more readable
```suggestion
type PairingPayloadManagerConfig struct {
	DB                   *multiaccounts.Database
	KeystorePath  string
	KeyUID           string
	Password        string
}
```

Reply:
😞 I want to be a beautiful snowflake